### PR TITLE
Revise Setup Tunnel Function

### PR DESCRIPTION
**Pull request by seanstrom**
Revise the `setupTunnel` function in `request.js`
Review? @nylen @simov @FredKSchott

### CODE REVIEW

File: request.js [hunk 1]
```diff
@@ -270,50 +270,35 @@ function debug() {
...
-  }
-
   var proxyHeaderWhiteList = self.proxyHeaderWhiteList.concat(proxyHeaderExclusiveList)
 
+  // Setup Proxy Headers and Proxy Headers Host
+  // Only send the Proxy White Listed Header names
   var proxyHost = constructProxyHost(self.uri)
-  self.proxyHeaders = constructProxyHeaderWhiteList(self.headers, proxyHeaderWhiteList)
+  var proxyHeaders = constructProxyHeaderWhiteList(self.headers, proxyHeaderWhiteList)
+  self.proxyHeaders = proxyHeaders
```
Comment (nylen) [inline request.js]: why split this into 2 lines?

**Comment by nylen** [comment]
Looks good to me (doesn't change behavior), I'm not totally clear on all the edge cases for this code though (i.e. is it possible to override headers in the default lists, are the default header lists and custom  lists all well-tested)

**Comment by seanstrom** [comment]
Yeah Im totally sure this is all tested, Im going to look through the tests and see where this is tested.
I believe the files should be either test-tunnel.js or test-proxy.js

Do you know?

**Comment by nylen** [comment]
`test-proxy-connect.js`, don't worry about it for this PR unless you see any issues.  I say this a lot, more tests are always a good thing :)

**Comment by seanstrom** [comment]
I accidentally forgot to say "NOT" when I said I was totally sure about something.
I want to add more tests, I was looking into the test coverage a bit and I am suspicious about the actual test coverage. It may not come into play this PR, but I am plan on doing an audit of sorts on Test Coverage.

And yes I totally agree with your opinion @nylen :)

**Pull request merged**